src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,300 @@
+     * @throws EmptyListException if the TaskList is empty
+     */
+    public String findFromList(String command) throws IllegalFormatException, EmptyListException {
+        String regexToMatch = "^find .*";
+        String correctFormat = "find <keyword to find>";
+        checkCommandFormat(command, regexToMatch, correctFormat);
+
+        String keyword = command.substring(5).trim();
+
+        List<Task> filteredTaskList = tasks.stream()
+                                           .filter(str -> str.toString().contains(keyword))
+                                           .collect(Collectors.toList());
+
+        return printTaskList(filteredTaskList);
+    }
+
+    /**
+     * Prints the full TaskList stored.
+     *
+     * @return string representation of full TaskList stored
+     * @throws EmptyListException if the TaskList is empty
+     */
+    public String printFullTaskList() throws EmptyListException {
+        return printTaskList(tasks);
+    }
+
+    /**
+     * Prints the TaskList stored so far.
+     *
+     * @return string representation of TaskList
+     * @throws EmptyListException if TaskList is empty
+     */
+    public String printTaskList(List<Task> tasks) throws EmptyListException {
+        String s = "";
+        int size = tasks.size();
+
+        if (size == 0) {
+            throw new EmptyListException();
+        }
+
+        for (int i = 0; i < size; i++) {
+            s = s.concat(tasks.get(i).toString());
+            if (i < tasks.size() - 1) {
+                s = s.concat("\n");
+            }
+        }
+        return s;
+    }
+
+    /**
+     * Generates message for successful addition of task.
+     *
+     * @param task task added to TaskList
+     * @param size size of TaskList after addition
+     * @return message for successful addition of task
+     */
+    private String addMessage(Task task, int size) {
+        return "Got it. I've added this task:\n" + task
+                + "\nNow you have " + size
+                + (size <= 1
+                  ? " task "
+                  : " tasks " + "in your list.");
+    }
+
+    /**
+     * Generates message for successful deletion of task.
+     *
+     * @param task task deleted from TaskList
+     * @param size size of TaskList after deletion
+     * @return message for successful deletion of task
+     */
+    private String deleteMessage(Task task, int size) {
+        return "Noted. I've removed the following task:\n" + task
+                + "\nNow you have " + size
+                + (size <= 1
+                  ? " task "
+                  : " tasks " + "in your list.");
+    }
+
+    /**
+     * Generates message for successfully marking task as done.
+     *
+     * @param task task from TaskList marked as done.
+     * @return message for successfully marking task as done
+     */
+    private String doneMessage(Task task) {
+        return "Nice! I've marked this task as done:\n" + task;
+    }
+}
```

Comment:
I like how descriptive your javadocs are! Maybe you have forgotten some punctuations?